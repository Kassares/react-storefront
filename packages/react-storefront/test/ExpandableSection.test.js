/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import ExpandableSection from '../src/ExpandableSection'
import TestProvider from './TestProvider'
import AppModelBase from '../src/model/AppModelBase'

describe('ExpandableSection', () => {
  it('should render', () => {
    const app = AppModelBase.create({ amp: false })

    expect(
      mount(
        <TestProvider app={app}>
          <ExpandableSection title="Title">Foo</ExpandableSection>
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render AmpExpandableSection when amp == true', () => {
    const app = AppModelBase.create({ amp: true })

    expect(
      mount(
        <TestProvider app={app}>
          <ExpandableSection title="Title">Foo</ExpandableSection>
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render expanded when expanded=true', () => {
    const app = AppModelBase.create({ amp: true })

    expect(
      mount(
        <TestProvider app={app}>
          <ExpandableSection title="Title" expanded>
            Foo
          </ExpandableSection>
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('should render expanded when defaultExpanded=true', () => {
    const app = AppModelBase.create({ amp: false })
    const wrapper = mount(
      <TestProvider app={app}>
        <ExpandableSection title="Title" defaultExpanded>
          Foo
        </ExpandableSection>
      </TestProvider>
    )
    expect(wrapper.getDOMNode().className.includes('MuiExpansionPanel-expanded')).toBe(true)
  })
})
