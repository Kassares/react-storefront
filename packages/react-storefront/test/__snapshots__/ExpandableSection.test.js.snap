// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableSection should render 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(WithTheme(inject-ExpandableSection)) title="Title">
        <WithTheme(inject-ExpandableSection) title="Title" classes={{...}}>
          <inject-ExpandableSection theme={{...}} title="Title" classes={{...}}>
            <ExpandableSection theme={{...}} title="Title" classes={{...}} amp={false} margins={true}>
              <WithStyles(ExpansionPanel) classes={{...}} expanded={[undefined]} defaultExpanded={[undefined]} onChange={[Function]}>
                <ExpansionPanel expanded={[undefined]} defaultExpanded={false} onChange={[Function]} classes={{...}} disabled={false}>
                  <WithStyles(Paper) className="MuiExpansionPanel-root-13 RSFExpandableSection-root-1 RSFExpandableSection-margins-2 RSFExpandableSection-expandedPanel-10" elevation={1} square={true}>
                    <Paper className="MuiExpansionPanel-root-13 RSFExpandableSection-root-1 RSFExpandableSection-margins-2 RSFExpandableSection-expandedPanel-10" elevation={1} square={true} classes={{...}} component="div">
                      <div className="MuiPaper-root-16 MuiPaper-elevation1-19 MuiExpansionPanel-root-13 RSFExpandableSection-root-1 RSFExpandableSection-margins-2 RSFExpandableSection-expandedPanel-10">
                        <WithStyles(ExpansionPanelSummary) expandIcon={{...}} classes={{...}} disabled={false} expanded={false} onChange={[Function]}>
                          <ExpansionPanelSummary expandIcon={{...}} disabled={false} expanded={false} onChange={[Function]} classes={{...}}>
                            <WithStyles(ButtonBase) focusRipple={false} disableRipple={true} disabled={false} component="div" aria-expanded={false} className="MuiExpansionPanelSummary-root-43" onFocusVisible={[Function]} onBlur={[Function]} onClick={[Function]}>
                              <ButtonBase focusRipple={false} disableRipple={true} disabled={false} component="div" aria-expanded={false} className="MuiExpansionPanelSummary-root-43" onFocusVisible={[Function]} onBlur={[Function]} onClick={[Function]} classes={{...}} centerRipple={false} disableTouchRipple={false} tabIndex="0" type="button">
                                <div className="MuiButtonBase-root-49 MuiExpansionPanelSummary-root-43" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" role="button" aria-expanded={false} onClick={[Function]}>
                                  <div className="MuiExpansionPanelSummary-content-47 RSFExpandableSection-summary-7">
                                    <WithStyles(Typography) variant="subtitle1">
                                      <Typography variant="subtitle1" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                        <h6 className="MuiTypography-root-52 MuiTypography-subtitle1-70">
                                          Title
                                        </h6>
                                      </Typography>
                                    </WithStyles(Typography)>
                                  </div>
                                  <WithStyles(IconButton) disabled={false} className="MuiExpansionPanelSummary-expandIcon-48 RSFExpandableSection-summaryIconWrap-9" component="div" tabIndex={-1} aria-hidden="true">
                                    <IconButton disabled={false} className="MuiExpansionPanelSummary-expandIcon-48 RSFExpandableSection-summaryIconWrap-9" component="div" tabIndex={-1} aria-hidden="true" classes={{...}} color="default">
                                      <WithStyles(ButtonBase) className="MuiIconButton-root-87 MuiExpansionPanelSummary-expandIcon-48 RSFExpandableSection-summaryIconWrap-9" centerRipple={true} focusRipple={true} disabled={false} component="div" tabIndex={-1} aria-hidden="true">
                                        <ButtonBase className="MuiIconButton-root-87 MuiExpansionPanelSummary-expandIcon-48 RSFExpandableSection-summaryIconWrap-9" centerRipple={true} focusRipple={true} disabled={false} component="div" tabIndex={-1} aria-hidden="true" classes={{...}} disableRipple={false} disableTouchRipple={false} type="button">
                                          <div className="MuiButtonBase-root-49 MuiIconButton-root-87 MuiExpansionPanelSummary-expandIcon-48 RSFExpandableSection-summaryIconWrap-9" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role="button" aria-hidden="true">
                                            <span className="MuiIconButton-label-92">
                                              <pure(ExpandMoreIcon) className="RSFExpandableSection-expandIcon-11">
                                                <ExpandMoreIcon className="RSFExpandableSection-expandIcon-11">
                                                  <WithStyles(SvgIcon) className="RSFExpandableSection-expandIcon-11">
                                                    <SvgIcon className="RSFExpandableSection-expandIcon-11" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                      <svg className="MuiSvgIcon-root-93 RSFExpandableSection-expandIcon-11" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                        <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                        <path fill="none" d="M0 0h24v24H0z" />
                                                      </svg>
                                                    </SvgIcon>
                                                  </WithStyles(SvgIcon)>
                                                </ExpandMoreIcon>
                                              </pure(ExpandMoreIcon)>
                                            </span>
                                            <NoSsr defer={false} fallback={{...}}>
                                              <WithStyles(TouchRipple) innerRef={[Function]} center={true}>
                                                <TouchRipple center={true} classes={{...}}>
                                                  <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-107" childFactory={[Function: childFactory]}>
                                                    <span className="MuiTouchRipple-root-107" />
                                                  </TransitionGroup>
                                                </TouchRipple>
                                              </WithStyles(TouchRipple)>
                                            </NoSsr>
                                          </div>
                                        </ButtonBase>
                                      </WithStyles(ButtonBase)>
                                    </IconButton>
                                  </WithStyles(IconButton)>
                                </div>
                              </ButtonBase>
                            </WithStyles(ButtonBase)>
                          </ExpansionPanelSummary>
                        </WithStyles(ExpansionPanelSummary)>
                        <WithStyles(Collapse) in={false} timeout="auto" aria-hidden="true">
                          <Collapse in={false} timeout="auto" aria-hidden="true" theme={{...}} classes={{...}} collapsedHeight="0px" component="div">
                            <Transition onEnter={[Function]} onEntered={[Function]} onEntering={[Function]} onExit={[Function]} onExiting={[Function]} addEndListener={[Function]} timeout={{...}} in={false} aria-hidden="true" mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onExited={[Function: noop]}>
                              <div className="MuiCollapse-container-102" style={{...}} aria-hidden="true">
                                <div className="MuiCollapse-wrapper-104">
                                  <div className="MuiCollapse-wrapperInner-105">
                                    <WithStyles(ExpansionPanelDetails) classes={{...}}>
                                      <ExpansionPanelDetails classes={{...}}>
                                        <div className="MuiExpansionPanelDetails-root-106 RSFExpandableSection-details-6">
                                          Foo
                                        </div>
                                      </ExpansionPanelDetails>
                                    </WithStyles(ExpansionPanelDetails)>
                                  </div>
                                </div>
                              </div>
                            </Transition>
                          </Collapse>
                        </WithStyles(Collapse)>
                      </div>
                    </Paper>
                  </WithStyles(Paper)>
                </ExpansionPanel>
              </WithStyles(ExpansionPanel)>
            </ExpandableSection>
          </inject-ExpandableSection>
        </WithTheme(inject-ExpandableSection)>
      </WithStyles(WithTheme(inject-ExpandableSection))>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`ExpandableSection should render AmpExpandableSection when amp == true 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(WithTheme(inject-ExpandableSection)) title="Title">
        <WithTheme(inject-ExpandableSection) title="Title" classes={{...}}>
          <inject-ExpandableSection theme={{...}} title="Title" classes={{...}}>
            <ExpandableSection theme={{...}} title="Title" classes={{...}} amp={true} margins={true}>
              <WithStyles(AmpExpandableSection) ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title">
                <AmpExpandableSection ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title" classes={{...}} expanded={false}>
                  <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                    <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                      <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                    </SideEffect(NullComponent)>
                  </HelmetWrapper>
                  <amp-accordion disable-session-states={true} class="RSFAmpExpandableSection-accordion-126">
                    <section className="RSFAmpExpandableSection-section-128">
                      <WithStyles(Typography) variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-127">
                        <Typography variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-127" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                          <h3 className="MuiTypography-root-133 MuiTypography-subtitle1-151 RSFAmpExpandableSection-title-127">
                            Title
                            <pure(ExpandMoreIcon) className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-expand-130">
                              <ExpandMoreIcon className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-expand-130">
                                <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-expand-130">
                                  <SvgIcon className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-expand-130" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-168 RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-expand-130" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                      <path fill="none" d="M0 0h24v24H0z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </ExpandMoreIcon>
                            </pure(ExpandMoreIcon)>
                            <pure(ExpandLessIcon) className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-collapse-131">
                              <ExpandLessIcon className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-collapse-131">
                                <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-collapse-131">
                                  <SvgIcon className="RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-collapse-131" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-168 RSFAmpExpandableSection-toggle-129 RSFAmpExpandableSection-collapse-131" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                      <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                      <path fill="none" d="M0 0h24v24H0z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </ExpandLessIcon>
                            </pure(ExpandLessIcon)>
                          </h3>
                        </Typography>
                      </WithStyles(Typography)>
                      <div className="RSFAmpExpandableSection-body-132">
                        Foo
                      </div>
                    </section>
                  </amp-accordion>
                </AmpExpandableSection>
              </WithStyles(AmpExpandableSection)>
            </ExpandableSection>
          </inject-ExpandableSection>
        </WithTheme(inject-ExpandableSection)>
      </WithStyles(WithTheme(inject-ExpandableSection))>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`ExpandableSection should render expanded when expanded=true 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(WithTheme(inject-ExpandableSection)) title="Title" expanded={true}>
        <WithTheme(inject-ExpandableSection) title="Title" expanded={true} classes={{...}}>
          <inject-ExpandableSection theme={{...}} title="Title" expanded={true} classes={{...}}>
            <ExpandableSection theme={{...}} title="Title" expanded={true} classes={{...}} amp={true} margins={true}>
              <WithStyles(AmpExpandableSection) ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title">
                <AmpExpandableSection ExpandIcon={[Function: ShouldUpdate]} CollapseIcon={[Function: ShouldUpdate]} title="Title" classes={{...}} expanded={false}>
                  <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                    <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                      <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
                    </SideEffect(NullComponent)>
                  </HelmetWrapper>
                  <amp-accordion disable-session-states={true} class="RSFAmpExpandableSection-accordion-189">
                    <section className="RSFAmpExpandableSection-section-191">
                      <WithStyles(Typography) variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-190">
                        <Typography variant="subtitle1" component="h3" className="RSFAmpExpandableSection-title-190" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                          <h3 className="MuiTypography-root-196 MuiTypography-subtitle1-214 RSFAmpExpandableSection-title-190">
                            Title
                            <pure(ExpandMoreIcon) className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-expand-193">
                              <ExpandMoreIcon className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-expand-193">
                                <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-expand-193">
                                  <SvgIcon className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-expand-193" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-231 RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-expand-193" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                      <path fill="none" d="M0 0h24v24H0z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </ExpandMoreIcon>
                            </pure(ExpandMoreIcon)>
                            <pure(ExpandLessIcon) className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-collapse-194">
                              <ExpandLessIcon className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-collapse-194">
                                <WithStyles(SvgIcon) className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-collapse-194">
                                  <SvgIcon className="RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-collapse-194" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-231 RSFAmpExpandableSection-toggle-192 RSFAmpExpandableSection-collapse-194" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                      <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                      <path fill="none" d="M0 0h24v24H0z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </ExpandLessIcon>
                            </pure(ExpandLessIcon)>
                          </h3>
                        </Typography>
                      </WithStyles(Typography)>
                      <div className="RSFAmpExpandableSection-body-195">
                        Foo
                      </div>
                    </section>
                  </amp-accordion>
                </AmpExpandableSection>
              </WithStyles(AmpExpandableSection)>
            </ExpandableSection>
          </inject-ExpandableSection>
        </WithTheme(inject-ExpandableSection)>
      </WithStyles(WithTheme(inject-ExpandableSection))>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
